server/svix-server/src/queue/mod.rs

```diff
@@ -150,26 +170,34 @@ impl TaskQueueProducer {
 
 pub enum TaskQueueConsumer {
     Redis(RedisQueueConsumer),
-    Memory(MemoryQueueConsumer),
     RabbitMq(rabbitmq::Consumer),
+    Omni(DynConsumer),
 }
 
 impl TaskQueueConsumer {
     pub async fn receive_all(&mut self) -> Result<Vec<TaskQueueDelivery>> {
         match self {
             TaskQueueConsumer::Redis(q) => q.receive_all().await.trace(),
-            TaskQueueConsumer::Memory(q) => q.receive_all().await.trace(),
             TaskQueueConsumer::RabbitMq(q) => q.receive_all().await.trace(),
+            // FIXME(onelson): need to figure out what deadline/duration to use here
```

Comment:
This looks like it also needs resolving before this can be merged.

Reply:
@svix-onelson any ideas on how to find a suitable duration?

Reply:
Yeah, this is a struggle. For the old memory backend it looks like there was a 30 sec window used. The rabbit impl we have sort of cheats and will wait as long as it takes for 1 message to come through, then yields that 1 immediately.

My concern is the polling semantics for each backend furnished by omniqueue might suggest a different value for the deadline. It might need to therefore be something we pull from the service config.

My suggestion is to keep it 30s to start with a mind to revisit and possibly add a config value to drive it.